trackers_integration/issuetracker/mantis.py

```diff
@@ -0,0 +1,159 @@
+        url = f"{self.base_url}/projects"
+        return self._request("GET", url, headers=self.headers)
+
+    def get_issue(self, issue_id):
+        url = f"{self.base_url}/issues/{issue_id}"
+        return self._request("GET", url, headers=self.headers)
+
+    def create_issue(self, body):
+        url = f"{self.base_url}/issues/"
+        return self._request("POST", url, headers=self.headers, json=body)
+
+    def update_issue(self, issue_id, body):
+        url = f"{self.base_url}/issues/{issue_id}"
+        return self._request("PATCH", url, headers=self.headers, json=body)
+
+    def get_comments(self, issue_id):
+        url = f"{self.base_url}/issues/{issue_id}"
+        return self._request("GET", url, headers=self.headers)["issues"][0]["notes"]
+
+    def add_comment(self, issue_id, body):
+        url = f"{self.base_url}/issues/{issue_id}/notes"
+        return self._request("POST", url, headers=self.headers, json=body)
+
+    def delete_comment(self, issue_id, note_id):
+        headers = {"Content-type": "application/json"}
+        url = f"{self.base_url}/issues/{issue_id}/notes/{note_id}"
+        return requests.request("DELETE", url, headers=headers, timeout=30)
+
+    @staticmethod
+    def _request(method, url, **kwargs):
+        return requests.request(method, url, timeout=30, **kwargs).json()
+
+
+class MantisThread(IntegrationThread):
+    """
+    Execute Mantis code in a thread!
+
+    Executed from the IssueTracker interface methods.
+
+    :meta private:
+    """
+
+    def post_comment(self):
+        comment_body = {
+            "text": markdown2html(self.text()),
+        }
+        self.rpc.add_comment(self.bug_id, comment_body)
+
+
+class Mantis(IssueTrackerType):
+    """
+    Support for Mantis. Requires:
+
+    :base_url: URL to Mantis Server - e.g. https://mantisbt.org
+    :api_password: Mantis API token
+
+    .. note::
+        You can leave the ``api_username`` field blank since
+        those are not used by issue tracker!
+    """
+
+    it_class = MantisThread
+
+    def _rpc_connection(self):
+        return MantisAPI(self.bug_system.base_url, self.bug_system.api_password)
+
+    def is_adding_testcase_to_issue_disabled(self):
+        return not (self.bug_system.base_url and self.bug_system.api_password)
+
+    def get_project_from_mantis(self, execution):
+        """
+        Returns the project from the actual Mantis instance.
+        Will try to match execution.run.plan.product.name, otherwise will
+        return the first found!
+
+        You may override this method if you want more control and customization,
+        see https://kiwitcms.org/blog/tags/customization/
+
+        .. versionadded:: 11.4
+        """
+        projects = self.rpc.get_projects()["projects"]
+        try:
+            project = next(
+                project
+                for project in projects
+                if project["name"] == execution.run.plan.product.name
+            )
```

Comment:
Let's try not using list comprehension expressions hidden inside the `next()` function. You can take the default, e.g. `result = projects[0]` and then iterate over `projects` with a standard for loop + the if condition for the name (compare it lower case). If there was no match `result` would not change and the method will return the default anyway. 

Also return `project` and then use the name (and/or whatever else is needed) in the calling method. This will give more flexibility when someone decides to override this method.